[ryanchua00] iP

ryanchua00 opened the pull request:
Roody Application
---
> "Your mind is for having ideas, not holding them." - David Allen

The Roody application helps frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPERFAST to use!

All you need to do is,

1. download it from [here](https://github.com/ryanchua00/ip/releases)
2. double click it
3. add your tasks from the **command line**
4. let it manage your tasks for you :yum:

And it is **FREE**!

Features:

- [X]  Managing Tasks
- [X]  Managing Events
- [ ]  Reminders! (Coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `constructor` for Roody method:

```
/**
  * Creates a chatbot with specified filepath to task data.
  * @param filepath The filepath to task data.
  */
public Roody(String filepath) {
    // Assumed no more than 100 tasks
    this.list = new ArrayList<Task>();
    this.ui = new Ui();
    this.storage = new Storage(filepath);
}
```

Commit:
Add Gradle support

ryanchua00 committed:
level-1:
- Added Roody.java
- Created input reading and class commands

ryanchua00 committed:
Level-2
- Add the ability to store whatever text entered by the user and display them back to the user when requested.

ryanchua00 committed:
Level-2: Fix
- Added "this." to object-specific fields

ryanchua00 committed:
Level-3
- Added Task class
- Added Functions to handle mark/unmark

ryanchua00 committed:
Level-3
- Improved speak function

ryanchua00 committed:
Level-4
- Added inheriting classes of task
- Changed Task to abstract

ryanchua00 committed:
Level-4
- Added toString() overrides

ryanchua00 committed:
A-TextUiTesting
- Changed bat file to keep prompt open

ryanchua00 committed:
Level-5
- Added exception class
- Handled exceptions

ryanchua00 committed:
Level-5
- Handled errors

ryanchua00 committed:
Level-6:
- Added delete functionality

peaol commented on src/main/java/Roody.java:
```diff
@@ -152,13 +163,25 @@ public static void main(String[] args){
             // else, repeat
             } else if (inputs[0].equals("list")) {
                 roody.printList();
+            } else if (inputs[0].equals("delete")) {
+                if (inputs.length == 2) {                    
+                    roody.delete(inputs[1]);
+                } else {
+                    new RoodyException("Please enter a index number to be deleted");
```
Maybe can try using switch statements instead of if-statements here to have a neater code 😄 
the method split() can be helpful in determining "delete", "list" or "mark" cases etc!

ryanchua00 replied:
Thanks for this, made it smoother.

peaol commented on src/main/java/Roody.java:
```diff
@@ -3,15 +3,13 @@
 import java.util.Scanner;
```
Nice, you followed the coding standards for import statements!

peaol reviewed (commented):
Good job on implementing Roody.java, perhaps adding some more OOP like classes would be good!

ryanchua00 committed:
Level-7
- Added Task.saveTask() method for ease of storage
- Added saveFile() and loadFile() methods

ryanchua00 committed:
Level-8
- Changed Strings to LocalDate

ryanchua00 committed:
Merge branch 'branch-Level-7'

ryanchua00 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java
#	src/main/java/Roody.java

ryanchua00 committed:
A-MoreOOP v1.0

ryanchua00 committed:
A-MoreOOP v2.0

ryanchua00 committed:
A-Packages

ryanchua00 committed:
Merge branch 'add-gradle-support'

ryanchua00 committed:
A-Gradle

ryanchua00 committed:
A-JUnit

ryanchua00 committed:
A-Jar

ryanchua00 committed:
A-Javadoc

ryanchua00 committed:
A-CodingStandard

ryanchua00 committed:
Level-9

lennoxtr commented on src/main/java/roody/Roody.java:
```diff
@@ -0,0 +1,130 @@
+    }
+    
+    // toggles completion status of tasks
+    private void complete(int index, boolean complete) throws RoodyException{
```
perhaps a javadoc @param comment here would be helpful to explain why boolean complete is needed

lennoxtr commented on src/main/java/roody/Roody.java:
```diff
@@ -0,0 +1,130 @@
+    }
+    
+
+    private void delete(int index) {
```
A more intuitive name here would be great
```suggestion
    private void deleteTask(int index) {
```

lennoxtr reviewed (commented):
Your code is intuitive and very easy to read. Good Job!

ryanchua00 committed:
Merge branch 'branch-A-JavaDoc'

ryanchua00 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/roody/Deadline.java
#	src/main/java/roody/Roody.java
#	src/main/java/roody/Storage.java
#	src/main/java/roody/Task.java
#	src/main/java/roody/Todo.java

ryanchua00 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/roody/Deadline.java
#	src/main/java/roody/Event.java
#	src/main/java/roody/Parser.java
#	src/main/java/roody/Roody.java
#	src/main/java/roody/Todo.java
#	src/main/java/roody/Ui.java

ryanchua00 committed:
Level-10 Base

ryanchua00 committed:
Added Graphic User Interface(GUI) functionality

Add GUI

Added classes to handle GUI and changed Roody.java to launch GUI.
Depreciated CLI and UI functions were either removed or converted to
a String-based message format to pass to the GUI.

Improve switch flow in Roody.java

Methods in switch cases were extracted to individual methods. Error
handling to be improved to consolidate try-catch blocks.

ryanchua00 committed:
Improve Exception handling

Handle exceptions in Parser.java

While Roody.java used to handle exceptions and execute methods,
exception handling has been shifted to Parser.java for ease of
understanding. Only parsed input is passed to Roody.java,
simplifying command flow.

Add more custom exceptions and constants

To reduce repeated strings and literals, custom exceptions with
 fixed messages have been added for easier throwing and usage.
Constants have also been added to reduce magic literals.
*Roody.java
*Parser.java

ryanchua00 committed:
A-Assertions

ryanchua00 committed:
Add custom commands

Commands to carry out individual tasks captured by abstract
Command.java class and inherited by child command classes.
This classes override execute method, allowing for a single method
to be called in Roody.java

Check input format and create command

Parser.java now checks format of commands first before creating the
relevant commands and passing it back to Roody.java. Roody.java now
does not know the exact command, only that it is a command.

ryanchua00 committed:
Merge pull request #1 from ryanchua00/branch-A-CodeQuality

Branch-A-code quality

ryanchua00 committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/roody/Parser.java
#	src/main/java/roody/Storage.java

ryanchua00 committed:
Merge pull request #3 from ryanchua00/branch-A-Assertions

Branch-A-Assertions

ryanchua00 committed:
Create gradle.yml

ryanchua00 committed:
Update gradle.yml

ryanchua00 committed:
Update Storage.java

According to workflow

ryanchua00 committed:
Fixing according to CI

ryanchua00 committed:
Fix for Ci

ryanchua00 committed:
Add help functionality

ryanchua00 committed:
Merge pull request #4 from ryanchua00/branch-C-Help

Add help functionality

ryanchua00 committed:
Improve Code Quality

Improved code quality according to issue posted by bot. Applied SLAP
and renamed relevant variables.

ryanchua00 committed:
Improve GUI

Added infinite dialog length and removed dead code.

ryanchua00 committed:
Add User Guide

Add Ui.png

Improve GUI

ryanchua00 committed:
Change photo in README

ryanchua00 committed:
Changed UI picture file name

ryanchua00 committed:
Changing name pt1

ryanchua00 committed:
Change Ui.png

ryanchua00 committed:
Added picture to Docs folder
